### DISCUSSION TITLE
vitest fails with MODULE_NOT_FOUND even though the module exists under node_modules

### DISCUSSION THREAD

**Discussion by Gorthog** [vitest-dev/vitest#4460] [Q&A] [answered]
![image](https://github.com/vitest-dev/vitest/assets/15230040/26247d93-aef4-4287-bff7-6f77466277d2)

I'm not sure but seems vitest is looking for the modules directly under the current folder (/scheduler) and not under node_modules.

I'm not using any configuration file, but even when i added vitest.config.ts with this content:

```ts
import { defineConfig } from 'vitest/config'

export default defineConfig({
  test: {
    deps: {
      moduleDirectories: ['node_modules', path.resolve('../../packages')],
    }
  },
})
```

nothing changed.

Any help will be appreciated.

**Comment by sheremet-va** [comment]
It uses root because it was not able to resolve `sequalize` dependency - is it mentioned in package.json?

**Comment by Gorthog** [comment]
Yes. It's also present under node_modules and running the code itself (dev code, not tests) which uses sequelize, is working

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by Gorthog** [comment, marked as answer]
Found the issue.
There was a symlink that pointed to a different project, and the missing npm module was in that folder (i was running in project /scheduler but there was a file that was symlinked between /scheduler and /backend, and vitest was looking for the module under /backend folder).
I'm not sure why vitest was looking for the module there, but once i installed all modules in /backend, everything worked fine.

**Comment by CarlosLiu-becls** [comment]
I have encountered a similar issue when I tried to run all unit tests via Test Explorer in VS IDE, and Copilot **reinstalls frontend dependencies (yarn/install)**, then the unit tests can be shown correctly, and below is the reason from Copilot:

### What actually happened
When Run All Tests is used in Visual Studio, it does test discovery across all supported test frameworks in the solution/workspace, not only Grpc.Net.Client.dll (xUnit/NUnit/MSTest).

Because src/WebApp contains frontend test tooling (vitest + jsdom), Visual Studio’s JavaScript/TypeScript test discovery also tried to initialize that test environment.

During that initialization, this chain loaded:
•	jsdom
•	data-urls
•	whatwg-url (transitive dependency)
In your node_modules, whatwg-url directory existed but its contents were missing/corrupted, so Node threw MODULE_NOT_FOUND.

### Why backend tests appeared blocked
Even if backend tests are unrelated, a discovery/adapter failure in another test platform can cause Test Explorer to show errors like:
•	no tests discovered
•	discovery aborted
•	run all failed
So it looked like backend tests depended on frontend modules, but the dependency was actually at the IDE test discovery pipeline level, not your backend code.

### Why reinstall fixed it
```yarn install``` rebuilt the corrupted frontend dependency tree, restoring whatwg-url files.
After that, JS test environment initialization no longer crashed, so Visual Studio could continue discovery and execution normally.
